### PR TITLE
[Merged by Bors] - feat: an ideal, with the operations induced from its parent ring, forms a non-unital subring

### PR DESCRIPTION
**Pull request by euprunin**
feat: an ideal, with the operations induced from its parent ring, forms a non-unital subring

Co-authored-by: Eric Wieser <wieser.eric@gmail.com>
Co-authored-by: Yakov Pechersky <pechersky@users.noreply.github.com>

---
This is my first PR, so please let me know if there are any improvements or adjustments needed to follow the project's conventions.

### CODE REVIEW

**Review by j-loreaux — COMMENTED** [review]
Can you please do the same for `TwoSidedIdeal`?

**Comment by j-loreaux** [comment]
Because I'm not sure about the new imports being too much ...

maintainer merge

File: Mathlib/RingTheory/TwoSidedIdeal/Basic.lean [hunk 1]
```diff
@@ -240,3 +243,8 @@ def subtypeMop : I →ₗ[Rᵐᵒᵖ] Rᵐᵒᵖ where
 end Ring
 
 end TwoSidedIdeal
+
+instance {R} [NonUnitalNonAssocRing R] : NonUnitalSubsemiringClass (TwoSidedIdeal R) R where
+  mul_mem _ hb := TwoSidedIdeal.mul_mem_left _ _ _ hb
+
+instance {R} [Ring R] : NonUnitalSubringClass (TwoSidedIdeal R) R where
```
Comment (j-loreaux) [inline Mathlib/RingTheory/TwoSidedIdeal/Basic.lean:250]: oops, just caught this.
```suggestion
instance {R} [NonUnitalNonAssocRing R] : NonUnitalSubringClass (TwoSidedIdeal R) R where
```

**Comment by mattrobball** [comment]
I think adding imports to add instances is a sign you need to put them in a new file.

File: Mathlib/RingTheory/TwoSidedIdeal/Basic.lean [hunk 2]
```diff
@@ -240,3 +243,8 @@ def subtypeMop : I →ₗ[Rᵐᵒᵖ] Rᵐᵒᵖ where
 end Ring
 
 end TwoSidedIdeal
+
+instance {R} [NonUnitalNonAssocRing R] : NonUnitalSubsemiringClass (TwoSidedIdeal R) R where
+  mul_mem _ hb := TwoSidedIdeal.mul_mem_left _ _ _ hb
```
Comment (j-loreaux) [inline Mathlib/RingTheory/TwoSidedIdeal/Basic.lean:248]: Because it's implied by the one below.
```suggestion
```

**Comment by j-loreaux** [comment]
I think you can put the `Ideal` stuff in `RingTheory.Ideal.Operations`, and for the `TwoSidedIdeal` just start a new file (maybe `RingTheory.Ideal.Instances` or something). It will likely get moved / absorbed in the near future as that API is developed.

**Review by j-loreaux — COMMENTED** [review]
Thanks!

bors merge

**Pull request closed without merging**